### PR TITLE
Improve cauchy convergence to handle quantities that converge to small values

### PR DESCRIPTION
**Pull request by ArneVoss**
## Proposed Changes
*Give a brief overview of your contribution here in a few sentences.*
 
I believe it would be helpful if the user can choose between relative or absolute values for cauchy convergence. 
Example: For the clean aircraft, the rolling moment coefficient MOMENT_X is close to zero and thus will never reach a relative cauchy convergence ->> dividing tiny numbers is not a good idea. Using absolute cauchy convergence is more robust in this case.

At the same time, I understand that a more complex input / more parameters might confuse users. An automatic switch, as propsed in this pull request, is a good compromise. The code now switches automatically to absolut cauchy convergence if the coefficient is < 0.1 to avoid the problem described above. The config file is unchanged. 

See short discussion here: https://www.cfd-online.com/Forums/su2/240148-cauchy-convergence-criteria-cmx-cmy-cmz.html.

## Related Work
*Resolve any issues (bug fix or feature request), note any related PRs, or mention interactions with the work of others, if any.*

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [x] I am submitting my contribution to the develop branch.
- [x] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags, or simply --warnlevel=2 when using meson).
- [ ] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

### CODE REVIEW

File: SU2_CFD/src/output/COutput.cpp [hunk 1]
```diff
@@ -850,6 +850,7 @@ bool COutput::Convergence_Monitoring(CConfig *config, unsigned long Iteration) {
   for (unsigned short iField_Conv = 0; iField_Conv < convFields.size(); iField_Conv++){
 
     bool fieldConverged = false;
+    su2double cauchyScalingFactor = 1.0;
```
Comment (pcarruscag) [inline SU2_CFD/src/output/COutput.cpp:853]:
```suggestion
```

File: SU2_CFD/src/output/COutput.cpp [hunk 2]
```diff
@@ -873,7 +874,17 @@ bool COutput::Convergence_Monitoring(CConfig *config, unsigned long Iteration) {
...
         newFunc[iField_Conv] = monitor;
-        cauchyFunc = fabs(newFunc[iField_Conv] - oldFunc[iField_Conv])/fabs(monitor);
+        
+        /*--- Automatically modify the scaling factor of relative Cauchy convergence for 
+        * coefficients that are close to zero. Example: For the clean aircraft, the rolling 
+        * moment coefficient MOMENT_X is close to zero and thus will never reach a relative 
+        * cauchy convergence ->> dividing tiny numbers is not a good idea. Using absolute 
+        * cauchy convergence is more robust in this case. ---*/
+        
+        if (fabs(monitor) <= 0.1) { cauchyScalingFactor = 0.1; }
```
Comment (pcarruscag) [inline SU2_CFD/src/output/COutput.cpp:884]:
```suggestion
```

File: SU2_CFD/src/output/COutput.cpp [hunk 3]
```diff
@@ -873,7 +874,17 @@ bool COutput::Convergence_Monitoring(CConfig *config, unsigned long Iteration) {
...
+        else { cauchyScalingFactor = fabs(monitor); }
```
Comment (pcarruscag) [inline SU2_CFD/src/output/COutput.cpp:885]:
```suggestion
```

File: SU2_CFD/src/output/COutput.cpp [hunk 4]
```diff
@@ -873,7 +874,17 @@ bool COutput::Convergence_Monitoring(CConfig *config, unsigned long Iteration) {
...
+
```
Comment (pcarruscag) [inline SU2_CFD/src/output/COutput.cpp:886]:
```suggestion
```

File: SU2_CFD/src/output/COutput.cpp [hunk 5]
```diff
@@ -873,7 +874,17 @@ bool COutput::Convergence_Monitoring(CConfig *config, unsigned long Iteration) {
...
+        /*--- Automatically modify the scaling factor of relative Cauchy convergence for 
+        * coefficients that are close to zero. Example: For the clean aircraft, the rolling 
+        * moment coefficient MOMENT_X is close to zero and thus will never reach a relative 
+        * cauchy convergence ->> dividing tiny numbers is not a good idea. Using absolute 
+        * cauchy convergence is more robust in this case. ---*/
+        
+        if (fabs(monitor) <= 0.1) { cauchyScalingFactor = 0.1; }
+        else { cauchyScalingFactor = fabs(monitor); }
+        
+        cauchyFunc = fabs(newFunc[iField_Conv] - oldFunc[iField_Conv])/cauchyScalingFactor;
```
Comment (pcarruscag) [inline SU2_CFD/src/output/COutput.cpp:887]:
```suggestion
        cauchyFunc = fabs(newFunc[iField_Conv] - oldFunc[iField_Conv])/fmax(fabs(monitor), 0.1);
```

File: SU2_CFD/src/output/COutput.cpp [hunk 6]
```diff
@@ -873,7 +874,17 @@ bool COutput::Convergence_Monitoring(CConfig *config, unsigned long Iteration) {
...
         oldFunc[iField_Conv] = newFunc[iField_Conv];
         newFunc[iField_Conv] = monitor;
-        cauchyFunc = fabs(newFunc[iField_Conv] - oldFunc[iField_Conv])/fabs(monitor);
+        
+        /*--- Automatically modify the scaling factor of relative Cauchy convergence for 
+        * coefficients that are close to zero. Example: For the clean aircraft, the rolling 
+        * moment coefficient MOMENT_X is close to zero and thus will never reach a relative 
+        * cauchy convergence ->> dividing tiny numbers is not a good idea. Using absolute 
+        * cauchy convergence is more robust in this case. ---*/
+
```
Comment (pcarruscag) [inline SU2_CFD/src/output/COutput.cpp:883]:
```suggestion

```

File: SU2_CFD/src/output/COutput.cpp [hunk 7]
```diff
@@ -873,7 +874,17 @@ bool COutput::Convergence_Monitoring(CConfig *config, unsigned long Iteration) {
 
         oldFunc[iField_Conv] = newFunc[iField_Conv];
         newFunc[iField_Conv] = monitor;
-        cauchyFunc = fabs(newFunc[iField_Conv] - oldFunc[iField_Conv])/fabs(monitor);
+
```
Comment (pcarruscag) [inline SU2_CFD/src/output/COutput.cpp:877]:
```suggestion

```

**Review by ArneVoss — COMMENTED** [review]
Thank you for your review and for shortening/simplifying the code! Do I need to agree to your propsal? How do I do this?

**Comment by ArneVoss** [comment]
OK, great, thank you!

**Pull request merged**